packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -103,6 +103,15 @@ export async function coerceParameter(
       await validateParam(spec, data, options);
       return result;
     }
+
+    // For a single param.array, Express query parser passes the single value
+    // as a string instead of array of string. We convert the string to array
+    // before passing it into AJV
+    if (OAIType === 'array' && typeof data === 'string') {
```

Comment:
@mrmodise lgtm but probably makes sense that @raymondfeng reviews this

Reply:
This is probably too open. I would like to only check for `query` parameters with primitive array, such as `string[]`, 'number[]` and `boolean[]`. The spec can give us the condition.

Reply:
Thanks will update accordingly I think I have seen such conditions when going through this functionality.

Reply:
> ```ts
> function coerceArray(data: string | string[], spec: ParameterObject) {
>   if (typeof data === 'string') {
>       return [data];
>   }
> 
>   return data;
> }
> ```

I was doing this on my latest change but I feel like it does not cater for boolean and number primitives since the type is already restricted to string | string[]. Will this be the right place to include other primitives?

Reply:
changes applied accordingly